.github/workflows/go.yml

```diff
@@ -0,0 +1,133 @@
+name: Go Code
+
+on:
+  push:
+    paths:
+    - '*.go'
+    - '.revive.toml'
+    - 'cmd/**'
+    - 'go.mod'
+    - 'go.sum'
+    - 'Makefile'
+    - 'pkg/**'
+    - '!pkg/webui/**'
+    - 'pkg/webui/**.go'
+    - 'tools/**'
+    - '!tools/mage/docs.go'
+    - '!tools/mage/js.go'
+    - '!tools/mage/proto.go'
+    - '!tools/mage/scripts/fix-grpc-gateway-names.sh'
+    - '!tools/mage/styl.go'
+    - '!tools/mage/translations.js'
+    - '!tools/mage/version.go'
+    - '!tools/mage/xx.js'
+
+  pull_request:
+    paths:
+    - '*.go'
+    - '.revive.toml'
+    - 'cmd/**'
+    - 'go.mod'
+    - 'go.sum'
+    - 'Makefile'
+    - 'pkg/**'
+    - '!pkg/webui/**'
+    - 'pkg/webui/**.go'
+    - 'tools/**'
+    - '!tools/mage/docs.go'
+    - '!tools/mage/js.go'
+    - '!tools/mage/proto.go'
+    - '!tools/mage/scripts/fix-grpc-gateway-names.sh'
+    - '!tools/mage/styl.go'
+    - '!tools/mage/translations.js'
+    - '!tools/mage/version.go'
+    - '!tools/mage/xx.js'
+
+jobs:
+  quality:
+    name: Code Quality
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+    - name: Initialize Go cache
```

Comment:
Shouldn't we do this after the checkout? Since we're relying on the hash of `go.sum`?